prog-o-meter.py

```diff
@@ -143,23 +152,36 @@ def add_day(self):
         self.days_remaining = self.GOAL - self.days
         self.completion_date = self.get_completion_date(self.days_remaining)
         self.canvas.itemconfig(self.rectangle_list[self.days-1], fill = "green")
+        self.canvas.itemconfig(self.encourage_text, text = "".join(self.encourage[self.new_no(self.current_greeting)]))
         update_days_file(self.filename, self.days)
         self.canvas.itemconfig(self.countdown_text, text = "".join(("You have ", str(self.days_remaining), " days left!\n\n", "If you code everyday, you will be done with this project on ", self.completion_date)))
-        if self.days >=self.GOAL:        # Disable add_day_button if goal have been reached 
-            self.add_day_button.config(state = "disabled")             
+        if self.days >=self.GOAL:        # Disable add_day_button if goal have been reached
+            self.add_day_button.config(state = "disabled")
+            self.canvas.itemconfig(self.greeting, text=("".join(("Congrats! ", self.username))))
+    def new_no(self, current_greeting):
+        """Allows to choose a new encouragement from the list each time the button is clicked so that the encouragements are not repeated.
 
+            Attributes:
+            current_greeting: A flag attribute that stores the previous value of the encouragement index chosen so that it does not conflict with the new one.
+            Returns: A new generated index that is not repeated
+        """
+        new = randint(0, 9)

```

Comment:
adding the same comment here, as on line 170 would be useful :)

Reply:
Yes, I'm using Atom for writing the code. I am able to view them as spaces instead of tabs.